src/Workspaces/Remote/Core/SolutionAssetProvider.cs

```diff
@@ -34,6 +34,26 @@ public SolutionAssetProvider(SolutionServices services)
         }
 
         public async ValueTask GetAssetsAsync(PipeWriter pipeWriter, Checksum solutionChecksum, Checksum[] checksums, CancellationToken cancellationToken)
+        {
+            // The responsibility is on us (as per the requirements of RemoteCallback.InvokeAsync) to Complete the
+            // pipewriter.  This will signal to streamjsonrpc that the writer passed into it is complete, which will
+            // allow the calling side know to stop reading results.
+            try
+            {
+                await GetAssetsWorkerAsync(pipeWriter, solutionChecksum, checksums, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception ex)
```

Comment:
>Exception [](http://example.com/codeflow?start=19&length=9)

Can we avoid rethrowing here (and losing stack trace and stack frames in dump)?

```
Exception? exception = null;
try
{
   await GetAssetsWorkerAsync(...);
}
catch (Exception e) when ((exception = e) is null)
{
  throw ExceptionUtilities.Unreachable;
}
finally
{
  await pipeWriter.CompleteAsync(exception).ConfigureAwait(false);
}
```

Reply:
dumb question.  i thought that was what naked `throw;` did.  It just continues throwing the exception upwards.  Is that not hte case?  if not, i can def change to the above (though would prefer in the next pr).

Reply:
`throw` unwinds the stack and then throws the exception. Filter does not unwind the stack if the condition evaluates to false. That's why we report NFW in filters so that the dump contains all frames starting with the true origin of the exception. `throw` changes the origin.

Reply:
sounds good.  i'll change this to the above pattern in the next PR :)

Reply:
Done in 779e15512bf.